src/main/java/org/folio/rest/impl/FinanceHelper.java

```diff
@@ -232,4 +292,27 @@ public void checkForCustomTransactionError(Throwable fail) {
     }
   }
 
+  /**
+   * Calculates remaining amount for encumbrance
+   * [remaining amount] = (allocated * allowableEncumbered) - (allocated - (unavailable + available)) - (encumbered + awaitingPayment)
+   *
+   * @param budget     processed budget
+   * @return remaining amount for encumbrance
+   */
+  private Double getBudgetRemainingAmountForEncumbrance(Budget budget) {
```

Comment:
Could you use Java Money API for calculations in order to take into account the specifics of the currency

Reply:
leveraged java money api